components.go

```diff
@@ -0,0 +1,101 @@
+package discordgo
+
+import (
+	"encoding/json"
+)
+
+// ComponentType is type of component.
+type ComponentType uint
+
+// MessageComponent types.
+const (
+	ActionsRowComponent ComponentType = iota + 1
```

Comment:
Please avoid introducing additional uses of `iota` for constants that depend on external values; it's better to list them out explicitly (even if we don't do this universally in the library at the moment). See https://github.com/bwmarrin/discordgo/commit/843c765ae3eeecaf8e9b808ea8e3da10ba600f91 for more reasoning.